test/Microsoft.Azure.ServiceBus.UnitTests/BrokeredMessageTests.cs

```diff
@@ -105,5 +106,76 @@ void CustomMessageIdGenerator()
 
             BrokeredMessage.SetMessageIdGenerator(null);
         }
+
+        public class When_querying_IsReceived_property
+        {
+            [Fact]
+            [DisplayTestMethodName]
+            void Should_return_false_for_message_that_was_not_sent()
+            {
+                var message = new BrokeredMessage();
+                message.Properties["dummy"] = "dummy";
+                Assert.False(message.IsReceived);
+            }
+
+            [Theory]
+            [DisplayTestMethodName]
+            [InlineData(ReceiveMode.ReceiveAndDelete)]
+            [InlineData(ReceiveMode.PeekLock)]
+            async Task Should_return_true_for_message_that_was_sent_and_received(ReceiveMode receiveMode)
+            {
+                var queueClient = QueueClient.CreateFromConnectionString(TestUtility.GetEntityConnectionString(Constants.NonPartitionedQueueName), receiveMode);
+                try
+                {
+                    await TestUtility.SendMessagesAsync(queueClient.InnerSender, 1);
+                    var receivedMessages = await TestUtility.ReceiveMessagesAsync(queueClient.InnerReceiver, 1);
```

Comment:
>ReceiveMessagesAsync [](start = 61, length = 20)

For PeekLockMode test, you may have to complete here, so as to not leave behind messages that are picked up by other tests, until we have a pertest purging of the queues